api/v1/helpers/verbs/putUtils.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * api/v1/helpers/verbs/putUtils.js
+ */
+'use strict'; // eslint-disable-line strict
+
+const u = require('./utils');
+const publisher = u.publisher;
+const event = u.realtimeEvents;
+const httpStatus = require('../../constants').httpStatus;
+const redisCache = require('../../../../cache/redisCache').client.cache;
+
+/**
+ * Generic helper to update a resource.
+ * If no value was provided for an field, clears that field by setting its
+ * value to null (or false for boolean fields).
+ *
+ * @param {Object} req From Express
+ * @param {Object} props From Express
+ * @param {Array} puttableFields From req
+ */
+function getPutPromise(req, props, puttableFields) {
+  const toPut = req.swagger.params.queryBody.value;
+  return u.findByKey(
+    props, req.swagger.params
+  )
+  .then((o) => u.isWritable(req, o))
+  .then((o) => {
+    const keys = Object.keys(puttableFields);
+    for (let i = 0; i < keys.length; i++) {
+      const key = keys[i];
+      if (toPut[key] === undefined) {
+        let nullish = null;
+        if (puttableFields[key].type === 'boolean') {
+          nullish = false;
+        } else if (puttableFields[key].enum) {
+          nullish = puttableFields[key].default;
+        }
+
+        o.set(key, nullish);
+
+        // take nullified fields out of changed fields
+        o.changed(key, false);
+      } else {
+        /**
+         * value may have changed. set changed to true to
+         * trigger checks in the model
+         */
+        o.changed(key, true);
+        o.set(key, toPut[key]);
+      }
+    }
+
+    return o.save();
+  });
+}
+
+/**
+ * Sends the udpated record back in the json response
+ * with status code 200.
+ *
+ * @param {Object} o The updated sequelize instance
+ * @param {Object} resultObj For logging
+ * @param {Object} req From Express
+ * @param {Object} res From Express
+ * @param {Object} props From Express
```

Comment:
"props" is not from express. It is the model on which the update operation is performed.